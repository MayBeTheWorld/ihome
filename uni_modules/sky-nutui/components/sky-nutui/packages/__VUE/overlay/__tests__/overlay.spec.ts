import { mount } from '@vue/test-utils';
import OverLay from '../index.vue';

test('should change z-index when using z-index prop', () => {
  const wrapper = mount(OverLay, {
    props: {
      visible: true,
      zIndex: 99
    }
  });
  const overlay: any = wrapper.find('.nut-overlay');
  expect(overlay.element.style.zIndex).toEqual('99');
});

test('should change animation duration when using duration prop', () => {
  const wrapper = mount(OverLay, {
    props: {
      visible: true,
      duration: 1
    }
  });

  const overlay: any = wrapper.find('.nut-overlay');
  expect(overlay.element.style.animationDuration).toEqual('1s');
});

test('prop overlay-class test', () => {
  const wrapper = mount(OverLay, {
    props: {
      visible: true,
      overlayClass: 'testclass'
    }
  });

  const overlay: any = wrapper.find('.nut-overlay');
  expect(overlay.classes()).toContain('testclass');
});

test('prop close-on-click-overlay test', async () => {
  const wrapper = mount(OverLay, {
    props: {
      visible: true,
      closeOnClickOverlay: false
    }
  });
  const overlay: any = wrapper.find('.nut-overlay');
  await overlay.trigger('click');
  expect(overlay.element.style.display).toEqual('');
});

test('event click test', async () => {
  const wrapper = mount(OverLay, {
    props: {
      visible: true,
      closeOnClickOverlay: true
    }
  });

  const overlay: any = wrapper.find('.nut-overlay');
  await overlay.trigger('click');
  expect(wrapper.emitted('click')).toBeTruthy();
  expect((wrapper.emitted('update:visible') as any)[0][0]).toBe(false);
});

test('slots test', () => {
  const wrapper = mount(OverLay, {
    props: {
      visible: true
    },
    slots: {
      default: '这里是正文'
    }
  });

  expect(wrapper.html()).toMatchSnapshot();
});
