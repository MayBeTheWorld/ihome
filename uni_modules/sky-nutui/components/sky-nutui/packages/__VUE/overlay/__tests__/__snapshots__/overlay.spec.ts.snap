// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`slots test 1`] = `
"<transition-stub>
  <view class=\\"nut-overlay\\" style=\\"animation-duration: 0.3s; z-index: 2000;\\">这里是正文</view>
</transition-stub>"
`;
